libarchive/archive_read_support_format_zip.c

```diff
@@ -1168,20 +1168,18 @@ zip_read_local_file_header(struct archive_read *a, struct archive_entry *entry,
 	archive_entry_set_atime(entry, zip_entry->atime, 0);
 
 	if ((zip->entry->mode & AE_IFMT) == AE_IFLNK) {
-		size_t linkname_length;
+		size_t linkname_length = (size_t)zip_entry->compressed_size;
```

Comment:
Nit: const to stay in line with the other `const` additions through the patch?

Reply:
yeah